lib/pkg.ml

```diff
@@ -139,13 +139,24 @@ let delegate p =
 let build_dir p =
   match p.build_dir with Some b -> Ok b | None -> Ok (Fpath.v "_build")
 
+let is_backup_file p =
+  let str = Fpath.filename p in
+  let len = String.length str in
+  len < 0
+  &&
+  let first = str.[0] in
+  let last = str.[len - 1] in
+  Char.equal last '~' || (Char.equal first '#' && Char.equal last '#')
+
 let find_file path name =
   let open Fpath in
   OS.Dir.contents path >>| fun files ->
   List.filter
     (fun file ->
-      let name_no_ext = to_string (normalize (rem_ext file)) in
-      String.equal name (String.Ascii.lowercase name_no_ext))
+      if is_backup_file file then false
```

Comment:
Might be worth extracting the filtering logic into some kind of `find_file : name_wo_ext: string -> string list -> string list` in a separate module and test it as well!

What do you think?

Reply:
Agreed, I will improve this.

Reply:
done